pkg/util/util.go

```diff
@@ -0,0 +1,74 @@
+package util
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// This function validates the URL, then tries to fetch the data with retries
+// and then reads and returns the data as []byte
+// Returns an error if the URL is invalid, or fetching the data failed or
+// if reading the response body fails.
+func GetURLData(urlString string, attempts int) ([]byte, error) {
+	// Validate URL
+	_, err := url.ParseRequestURI(urlString)
+	if err != nil {
+		return nil, fmt.Errorf("invalid URL: %v", urlString)
+	}
+
+	// Fetch the URL and store the response body
+	body, err := FetchURLWithRetries(urlString, attempts, time.Second)
+	if err != nil {
+		return nil, fmt.Errorf("failed fetching data from the URL %v: %s", urlString, err)
+	}
+	defer body.Close()
+
+	// Read from the response body, ioutil.ReadAll will return []byte
+	data, err := ioutil.ReadAll(body)
+	if err != nil {
+		return nil, fmt.Errorf("reading from response body failed: %s", err)
+	}
+
+	return data, nil
+}
+
+// Try to fetch the given url string, and make <attempts> attempts at it.
+// Wait for <duration> time between each try.
+// This returns the response body upon successful fetch
+// The passed URL is not validated, so validate the URL before passing to this
+// function
+func FetchURLWithRetries(url string, attempts int, duration time.Duration) (io.ReadCloser, error) {
+	var body io.ReadCloser
+	var err error
+
+	for i := 0; i < attempts; i++ {
+		var response *http.Response
+
+		// sleep for <duration> seconds before trying again
+		if i > 0 {
+			time.Sleep(duration)
+		}
+
+		// retry if http.Get fails
+		// if all the retries fail, then return statement at the end of the
+		// function will return this err received from http.Get
+		response, err = http.Get(url)
+		if err != nil {
+			continue
+		}
+
+		// if the status code is not 200 OK, return an error
+		if response.StatusCode != http.StatusOK {
```

Comment:
@tnozicka while going through the entire [200 status list](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#2xx_Success), I'm not sure if all them can be treated as OK.
Also, Kubernetes for reference - https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/resource/visitor.go#L269

I partially agree with you, but I don't think we should do this for the entire 200 family. Let me know what should we go ahead with this, I'm a bit divided on this.